S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java

```diff
@@ -357,12 +366,52 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
     }
 
+
+    private void  requestPermission()
+    {
+        //ask for the permission in android M
+        int permission = ContextCompat.checkSelfPermission(this,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE);
+
+        if (permission != PackageManager.PERMISSION_GRANTED) {
+            Log.i(TAG, "Permission to record denied");
+
+            if (ActivityCompat.shouldShowRequestPermissionRationale(this,
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(this);
+                builder.setMessage("Permission to access the SD-CARD is required for this app to Download PDF.")
+                        .setTitle("Permission required");
+
+                builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
+
+                    public void onClick(DialogInterface dialog, int id) {
+                        Log.i(TAG, "Clicked");
+                        makeRequest();
+                    }
+                });
+
+                AlertDialog dialog = builder.create();
+                dialog.show();
+
+            } else {
+                makeRequest();
+            }
+        }
```

Comment:
Add a else block and add a log statement saying that: 
```
Log.i(TAG, "Permission to store data in external storage is granted.");
```

Reply:
done